signer/src/stacks_api.rs

```diff
@@ -40,154 +36,324 @@ impl StacksClient {
     /// StacksSettings.
     pub fn new(settings: StacksSettings) -> Self {
         Self {
-            api_endpoint: settings.api.endpoint,
             node_endpoint: settings.node.endpoint,
             client: reqwest::Client::new(),
         }
     }
 
-    /// Get Stacks block IDs given the bitcoin block hash. Uses the Stacks API
-    /// via the GET /extended/v2/burn-blocks/:height_or_hash endpoint.
-    ///
-    /// See https://docs.hiro.so/api/get-burn-block
-    async fn get_block_ids(&self, hash: &bitcoin::BlockHash) -> Result<Vec<StacksBlockId>, Error> {
-        // The Stacks API expects the hash to be hex encoded with the
-        // leading 0x in the string, which is not produced by the Display
-        // implementation of bitcoin::hashes::sha256d::Hash (but is for
-        // the Debug implementation).
-        let hash: &bitcoin::hashes::sha256d::Hash = hash.as_raw_hash();
-        let path = format!("/extended/v2/burn-blocks/0x{}", hash);
-        let url = self.api_endpoint.join(&path).map_err(Error::PathParse)?;
-
-        tracing::debug!(%hash, "Fetching block IDs confirmed by bitcoin block from stacks API");
+    /// Fetch the raw stacks nakamoto block from a Stacks node given the
+    /// Stacks block ID.
+    async fn get_block(&self, block_id: StacksBlockId) -> Result<NakamotoBlock, Error> {
```

Comment:
To get some more helpful information in the logs, including the function name and block ID.
```suggestion
    /// Stacks block ID.
    [tracing::instrument(skip(self))]
    async fn get_block(&self, block_id: StacksBlockId) -> Result<NakamotoBlock, Error> {
```

Reply:
We have a chat scheduled for later today where we'll probably go over this but I wanted to post the spark notes here:
1. I'm generally okay with using `tracing::instrument` spans. Its provides a good dev experience and makes logging and debugging better when used well.
3. Recently, I've been going in the opposite direction, where I enter into as few tracing spans as possible (which is usually just 1 or 2). Instead, I've gotten enough context by including the necessary information during the log and by entering into a created span. I've liked this approach more.
4. I've also tried leaning into using [opentelemetry](https://opentelemetry.io/) for tracing things in a distributed setting. But I'm not 100% happy with this alternative setup but it's been good.
5. I don't think application operators can really benefit from opentelemetry (over using just tracing) unless other services use it as well. I think that is probably going to be the case for us with the signer and the stacks-node.

Reply:
Right we did not cover this part in our chat. Personally I favor using `tracing::instrument` because it will capture the function call stack (for every instrumented function) in addition to any parameters we choose to include in the context. That is very helpful if a single function is called from many places (which may not be the case for `get_block` though).

That being said, I am okay with either.

Reply:
> and by entering into a created span

Do you mean that we should explicitly create spans at select points in the code? That sounds interesting. I feel like if the code is broken down to isolated functions there wouldn't be a natural point to enter a span which isn't also another function call but I might be wrong here.

Reply:
> > and by entering into a created span
> 
> Do you mean that we should explicitly create spans at select points in the code? 

Yes, that is what I mean. But the key point is that you need to enter into as few tracing spans as possible.

My logic goes like this:
1. Having trace and span IDs in our logs allows us to easily track the code path for a discrete unit of work, such as receiving a request or some other triggered event. This is especially useful when we can have concurrent execution of a particular code path.
2. We want these trace and span IDs placed in the top level of our JSON logs so that it is picked up by tools like grafana. So placing them within the `tracing` span list was undesirable.
3. In tracing, entering into a new span clobbers/removes much of the metadata from prior spans at the top level of the JSON log.

It could very well be that I'm misremembering something or that I misconfigured something when trying to get trace and span IDs. And it could also be the case that this is unnecessary or undesirable for some reason within the signer.

Anyway, this is the reason that I tend to shy away from `tracing::instrument` nowadays. Before, I would give the exact same feedback during code review encouraging the use of `tracing::instrument`.